tasks/test_results_processor.py

```diff
@@ -148,6 +138,116 @@ def process_individual_upload(
+        # Obviously this is a very rough estimate of sizes. We are interested more
+        # in the difference between the insert approaches. SO this should be fine.
+        # And these aux memory structures take the bulk of extra memory we need
+        memory_used = getsizeof(parsed_testruns) // 1024
+        metrics.gauge(
+            key="test_results.processor.write_to_db.aux_memory_used",
+            value=memory_used,
+            unit="kilobytes",
+            tags=metric_tags,
+        )
+
+    def _bulk_write_tests_to_db(
+        self,
+        db_session: Session,
+        repoid: int,
+        upload_id: int,
+        parsed_testruns: List[Testrun],
+        flags_hash: str,
+    ):
+        metric_tags = {"method": "bulk_insert"}
+        memory_used = getsizeof(parsed_testruns) // 1024
+        with metrics.timing(key="test_results.processor.write_to_db", tags=metric_tags):
+
+            test_data = []
+            test_instance_data = []
+            for testrun in parsed_testruns:
+                # Build up the data for bulk insert
+                name = testrun.name
+                testsuite = testrun.testsuite
+                outcome = str(testrun.outcome)
+                duration_seconds = testrun.duration
+                failure_message = testrun.failure_message
+                test_id = generate_test_id(repoid, testsuite, name, flags_hash)
+
+                test_data.append(
+                    dict(
+                        id=test_id,
+                        repoid=repoid,
+                        name=name,
+                        testsuite=testsuite,
+                        flags_hash=flags_hash,
+                    )
+                )
+
+                test_instance_data.append(
+                    dict(
+                        test_id=test_id,
+                        upload_id=upload_id,
+                        duration_seconds=duration_seconds,
+                        outcome=outcome,
+                        failure_message=failure_message,
+                    )
+                )
+
+            # Save Tests
+            insert_on_conflict_do_nothing = (
+                insert(Test.__table__).values(test_data).on_conflict_do_nothing()
+            )
+            db_session.execute(insert_on_conflict_do_nothing)
+            db_session.flush()
+            # Save TestInstances
+            insert_test_instances = insert(TestInstance.__table__).values(
+                test_instance_data
+            )
+            db_session.execute(insert_test_instances)
+            db_session.flush()
+        # Memory outside the time metrics to not disturb the counter
+        # Obviously this is a very rough estimate of sizes. We are interested more
+        # in the difference between the insert approaches. SO this should be fine.
+        # And these aux memory structures take the bulk of extra memory we need
+        memory_used += getsizeof(test_data) // 1024
```

Comment:
I tested this out locally and the memory_used i was getting was 0, I'm not sure if this is just a local thing I ran into or if there's a problem with this approach, I'm okay with trying it out to see if it works out in prod, since that's just my experience running it locally

Reply:
Depending on the number of tests that you tested against it would be small enough in size that it would return 0. After all it's Kb and the integer division will round down.

Before adding the `// 1024` I was getting value for those calls, so... maybe that 🤷